src/main/java/com/fasterxml/jackson/databind/util/EnumValues.java

```diff
@@ -118,14 +118,52 @@ public static EnumValues constructFromToString(MapperConfig<?> config, Class<Enu
         return construct(config, enumClass, external);
     }
 
+    /**
+     * Returns String serializations of Enum name using an instance of {@link EnumNamingStrategy}.
+     * <p>
+     * The output {@link EnumValues} should contain values that are symmetric to
+     * {@link EnumResolver#constructUsingEnumNamingStrategy(DeserializationConfig, AnnotatedClass, EnumNamingStrategy)}.
+     *
+     * @since 2.16
+     */
+    public static EnumValues constructUsingEnumNamingStrategy(MapperConfig<?> config, AnnotatedClass annotatedClass,
+            EnumNamingStrategy namingStrategy)
+    {
+        // prepare data
+        final AnnotationIntrospector ai = config.getAnnotationIntrospector();
+        final Class<?> enumCls0 = annotatedClass.getRawType();
+        final Class<Enum<?>> enumCls = _enumClass(enumCls0);
+        final Enum<?>[] enumConstants = _enumConstants(enumCls0);
+
+        // introspect
+        String[] names = new String[enumConstants.length];
+        if (ai != null) {
+            ai.findEnumValues(config, annotatedClass, enumConstants, names);
+        }
+
+        // build
+        SerializableString[] textual = new SerializableString[enumConstants.length];
+        for (int i = 0, len = enumConstants.length; i < len; i++) {
+            Enum<?> enumValue = enumConstants[i];
+            String name = names[i];
+            if (name == null) {
+                name = namingStrategy.convertEnumToExternalName(enumValue.name());
+            }
+            textual[i] = config.compileString(name);
```

Comment:
```suggestion
            if (config.isEnabled(EnumFeature.WRITE_ENUMS_TO_LOWERCASE)) {
                name = name.toLowerCase();
            }
            textual[i] = config.compileString(name);
```

Should `WRITE_ENUMS_TO_LOWERCASE` apply here?

Reply:
I don't think there is a lowercase SerializationFeature for the general case

Reply:
Correct: lower-casing is datatype-specific, no `SerializationFeature` (nor will there ever be).

The main question is interesting one; I can see arguments either way.
I _think_ that since `EnumFeature.WRITE_ENUMS_TO_LOWERCASE` is slightly more dynamic (Enum naming strategy being defined as annotations in class definition or mix-in), it should have precedence and, when enabled, force lower-casing.

Reply:
So I think code as-is does what is my leaning. Great!

Reply:
@cowtowncoder the current code *doesn't* force lower-casing, I think you're looking at the suggested diff

Reply:
@iProdigy Ok I must have thought suggestion was applied and not looked at actual change. Thank you for following through, I just merged follow-up PR!